lib/bundler_api/web.rb

```diff
@@ -273,6 +273,24 @@ def get_cached_dependencies
     dependencies
   end
 
+  def get_cached_info(name)
+    key = "info/#{name}"
+    @dalli_client.get(key) || begin
+      calc_info = @gem_info.info(name)
+      @dalli_client.set(key, calc_info)
+      calc_info
+    end
+  end
+
+  def get_cached_names
+    key = "names"
+    @dalli_client.get(key) || begin
```

Comment:
this should use `fetch` passing in a block that'll get called if the key isn't set